distributed/dashboard/scheduler.py

```diff
@@ -292,6 +292,24 @@ def collect(self):
             sum, (tp.states for tp in self.server.task_prefixes.values())
         )
 
+        suspicious_tasks = CounterMetricFamily(
+            "dask_scheduler_tasks_suspicious",
+            "Total number of times a task has been marked suspicious",
+            labels=["name"],
+        )
+
+        suspicious_by_prefix = {
+            prefix.name: 0 for prefix in self.server.task_prefixes.values()
```

Comment:
Rather than pre-initializing a dictionary here with keys corresponding to every task, I think a `collections.Counter` is well suited to this

Reply:
Thanks for the review! 
According to [Robust Perception](https://www.robustperception.io/existential-issues-with-metrics) and the [Prometheus best practices](https://prometheus.io/docs/practices/instrumentation/#avoid-missing-metrics) timeseries should be initialised with zero.

> Where this usually comes up is for metrics with labels. It is recommended to initialise metrics to zero when your application starts, so that when there's an increment on a subsequent scrape we'll be able to see that the value has increased over time.

Reply:
Ok, cool.  I think a `Counter` still makes sense here and avoids the (admittedly small) cost of iterating over every task twice.

Maybe something like

```
from collections import Counter

suspicious_by_prefix = Counter()
for task in self.server.tasks.values():
    suspicious_by_prefix[task.prefix.name] += task.suspicious
```

Reply:
I think the counter is really cool, but what benefit does it give here over a regular dictionary?

I fully agree with the need to simplify the whole collection of suspicious tasks, but I currently don't see the need for a special data structure, which extends the dict with an interface we would not really use here.

Reply:
Oh nevermind. Thanks a lot for the advice regarding the Counter.